include/enums.h

```diff
@@ -1756,396 +1756,7 @@ enum AuxCameraParams {
-    ITEM_ATTACK_FX_C                  = 0x00000148,
-    ITEM_ATTACK_FX_F                  = 0x00000149,
-    ITEM_HP_PLUS_C                    = 0x0000014A,
-    ITEM_HP_PLUS_X                    = 0x0000014B,
-    ITEM_HP_PLUS_Y                    = 0x0000014C,
-    ITEM_FP_PLUS_C                    = 0x0000014D,
-    ITEM_FP_PLUS_X                    = 0x0000014E,
-    ITEM_FP_PLUS_Y                    = 0x0000014F,
-    ITEM_HEALTHY_HEALTHY              = 0x00000150,
-    ITEM_ATTACK_FX_F2                 = 0x00000151,
-    ITEM_ATTACK_FX_F3                 = 0x00000152,
-    ITEM_ATTACK_FX_F4                 = 0x00000153,
-    ITEM_ATTACK_FX_F5                 = 0x00000154,
-    ITEM_PARTNER_ATTACK               = 0x00000155,
-    ITEM_HEART                        = 0x00000156,
-    ITEM_COIN                         = 0x00000157,
-    ITEM_HEART_PIECE                  = 0x00000158,
-    ITEM_STAR_POINT                   = 0x00000159,
-    ITEM_HEART_POINT                  = 0x0000015A,
-    ITEM_FLOWER_POINT                 = 0x0000015B,
-    ITEM_STAR_PIECE                   = 0x0000015C,
-    ITEM_PRESENT                      = 0x0000015D,
-    ITEM_CAKE_DONE                    = 0x0000015E,
-    ITEM_CAKE_BARE                    = 0x0000015F,
-    ITEM_CAKE_PAN                     = 0x00000160,
-    ITEM_CAKE_BATTER                  = 0x00000161,
-    ITEM_CAKE_BOWL                    = 0x00000162,
-    ITEM_CAKE_MIXED                   = 0x00000163,
-    ITEM_CAKE_WITH_ICING              = 0x00000164,
-    ITEM_CAKE_WITH_BERRIES            = 0x00000165,
-    ITEM_HAMMER1_ICON                 = 0x00000166,
-    ITEM_HAMMER2_ICON                 = 0x00000167,
-    ITEM_HAMMER3_ICON                 = 0x00000168,
-    ITEM_BOOTS1_ICON                  = 0x00000169,
-    ITEM_BOOTS2_ICON                  = 0x0000016A,
-    ITEM_BOOTS3_ICON                  = 0x0000016B,
-    ITEM_ITEMS_ICON                   = 0x0000016C,
-};
-
-#define LOOKUP_ITEM(itemID) gItemTable[itemID & ~0xF0000]
-
-// item ranges:
-// 001 - 007 = gear (hammer/boots)
-// 008 - 00F = 'quest' items
-// 010 - 07F = keys
-// 080 - 0DA = consumables
-// 0DB - 0DF = unused
-// 0E0 - 154 badges
-// 155+      misc (collectables, utility items for menu purposes, etc)
-
-#define ITEM_FIRST_KEY              ITEM_KOOPA_FORTRESS_KEY
-#define ITEM_LAST_KEY               ITEM_PRISON_KEY4
-#define ITEM_FIRST_CONSUMABLE       ITEM_FIRE_FLOWER
-#define ITEM_LAST_CONSUMABLE        ITEM_JELLY_SHROOM1
-#define ITEM_FIRST_BADGE            ITEM_SPIN_SMASH
-#define ITEM_LAST_BADGE             ITEM_ATTACK_FX_F5
-
-#define ITEM_NUM_KEYS (ITEM_LAST_KEY - ITEM_FIRST_KEY + 1)
-#define ITEM_NUM_CONSUMABLES (ITEM_LAST_CONSUMABLE - ITEM_FIRST_CONSUMABLE + 1)
-#define IS_ITEM(itemID) (itemID >= ITEM_FIRST_KEY && itemID <= ITEM_LAST_CONSUMABLE)
-#define IS_BADGE(itemID) (itemID >= ITEM_FIRST_BADGE && itemID <= ITEM_LAST_BADGE)
+#include "gen/item_enum.h"
```

Comment:
I'm not sure we need to put generated includes/headers in a "gen" directory. For the effect stuff I recently did, I didn't do that. so if we do want to, we should probably be consistent..but I'd vote to just put this not in one